src/helpers/devserver.ts

```diff
@@ -0,0 +1,87 @@
+import fetch from "cross-fetch";
+import { envKeys, prodEnvKeys } from "./consts";
+import type { FunctionConfig } from "../types";
+
+export const available = async (): Promise<boolean> => {
+  if (isProd()) {
+    return false;
+  }
+
+  try {
+    const url = devserverURL("/dev");
+    const result = await fetch(url.toString());
+    await result.json();
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+// isProd compares any supported standard env variable for "production",
+// returning true on first match.
+export const isProd = (): boolean => {
+  return !!Object.values(prodEnvKeys).find(
+    (e) => process.env[e] === "production"
+  );
+};
+
+// url returns the dev server URL, overriding to use the INNGEST_DEVSERVER_URL
+// env var if provided.
+export const devserverURL = (pathname?: string): URL => {
+  let host = process.env[envKeys.DevServerURL];
+  if (!host) {
+    // Use the default.
+    const url = new URL(`http://127.0.0.1:8288/`);
+    url.pathname = pathname || "";
+    return url;
+  }
+
+  // Normalize the URL to be friendly here.  If the user hasn't added a scheme,
+  // default to http.
+  if (host.indexOf("://") === -1) {
+    host = `http://${host}`;
+  }
+
+  const url = new URL(host);
+  url.pathname = pathname || "";
+  return url;
+};
```

Comment:
Pending `process.env` change, could this be more succinct?

```suggestion
export const devServerUrl = (pathname = ""): URL => {
  let host =
    process.env.INNGEST_DEVSERVER_URL?.trim() || "http://127.0.0.1:8288/";
  if (!host.includes("://")) {
    host = `http://${host}`;
  }

  return new URL(pathname, host);
};
```